### PR TITLE
samples: sensor_shell: add fake sensor driver & pytest

### PR DESCRIPTION
**Pull request by ycsin**
Add a fake sensor driver so that the sample can be played with in a more meaningful way.

Then, use pytest to test the shell commands.

[Sample doc preview](https://builds.zephyrproject.io/zephyr/pr/72833/docs/samples/sensor/sensor_shell/README.html)

### CODE REVIEW

**Review by teburd — COMMENTED** [review]
Quite happy to see the testing here, the fake sensor makes sense to me.

File: samples/sensor/sensor_shell/src/fake_sensor.c
Comment (ubieda) [inline samples/sensor/sensor_shell/src/fake_sensor.c:1]: I wonder if there's potential synergy between this `fake_sensor` and `dummy_sensor` from #72052.

**Review by ubieda — DISMISSED** [review]

**Review by teburd — DISMISSED** [review]
Tried it locally works great, very nice thank you for adding this, its much needed.

File: samples/sensor/sensor_shell/sample.yaml [hunk 1]
```diff
@@ -11,3 +11,16 @@ tests:
...
+  sample.sensor.shell.pytest:
+    filter: CONFIG_SERIAL and dt_chosen_enabled("zephyr,shell-uart")
+    min_ram: 40
+    harness: pytest
+    extra_configs:
+      - arch:posix:CONFIG_NATIVE_UART_0_ON_STDINOUT=y
+      - CONFIG_SAMPLES_SENSOR_SHELL_FAKE_SENSOR=y
+    integration_platforms:
+      - qemu_riscv64
+      - native_posix
```
Comment (jeppenodgaard) [inline samples/sensor/sensor_shell/sample.yaml:23]: Any reason this is not `native_sim`?

File: samples/sensor/sensor_shell/src/fake_sensor.c
Comment (ycsin) [reply samples/sensor/sensor_shell/src/fake_sensor.c:1]: Had a brief look at #72052 & I reckon the `fake_sensor` in this PR can be updated to achieve the functionalities of the `dummy_sensor`, the testing in #72052 can also be tested in the pytest script with some changes and do not require ZTEST & the dummy shell backend, more 'organic' IMO.

cc @swenzel-arc

File: samples/sensor/sensor_shell/sample.yaml (hunk 1 above)
Comment (ycsin) [reply samples/sensor/sensor_shell/sample.yaml:23]: `native_posix` is what I always use for simple software stuff like this one, `qemu_riscv64` is what I use when `native` doesn't compile (MacOS), `riscv` is the arch that I work with on a daily basis. Just out-of-habit I guess

Comment (ycsin) [reply samples/sensor/sensor_shell/sample.yaml]: Updated to use `native_sim`, since that seems to be the direction

**Comment by ycsin** [comment]
Updated the fake driver to handle 3-axis data properly:
```
uart:~$ sensor get sensor@0 4
channel idx=7 gyro_xyz shift=0 num_samples=0 value=4294967295ns, (-0.000000, -0.000000, -0.000000)
[870:26:48.134,000] <inf> fake_sensor: [sample_fetch] dev: 0x8000e2b8, chan: 59
[870:26:48.134,000] <inf> fake_sensor: [channel_get] dev: 0x8000e2b8, chan: 4
uart:~$ sensor get sensor@0 7
channel idx=7 gyro_xyz shift=3 num_samples=1 value=9220133425350000000ns, (7.000000, 7.000000, 7.000000)
[870:26:48.134,000] <inf> fake_sensor: [sample_fetch] dev: 0x8000e2b8, chan: 59
[870:26:48.134,000] <inf> fake_sensor: [channel_get] dev: 0x8000e2b8, chan: 7
uart:~$ sensor get sensor@0 11
channel idx=11 magn_xyz shift=4 num_samples=1 value=9220133425350000000ns, (11.000000, 11.000000, 11.000000)
[870:26:48.134,000] <inf> fake_sensor: [sample_fetch] dev: 0x8000e2b8, chan: 59
[870:26:48.134,000] <inf> fake_sensor: [channel_get] dev: 0x8000e2b8, chan: 11
uart:~$ sensor get sensor@0 12
channel idx=12 die_temp shift=4 num_samples=1 value=9220133425350000000ns (12.000000)
[870:26:48.134,000] <inf> fake_sensor: [sample_fetch] dev: 0x8000e2b8, chan: 59
[870:26:48.134,000] <inf> fake_sensor: [channel_get] dev: 0x8000e2b8, chan: 12
```

File: samples/sensor/sensor_shell/sample.yaml (hunk 1 above)
Comment (ycsin) [reply samples/sensor/sensor_shell/sample.yaml:27]: `native_sim` seems to go bonkers in CI..

Comment (ycsin) [reply samples/sensor/sensor_shell/sample.yaml:27]: It went bonkers after #71093, probably not `native_sim` related

Comment (ycsin) [reply samples/sensor/sensor_shell/sample.yaml:27]: The sensor shell now stuck in an infinite loop when doing `sensor get <sensor> <channel>`, @teburd do you have any idea why?

Comment (teburd) [reply samples/sensor/sensor_shell/sample.yaml]: doh, thanks for finding and fixing, I think this is what @ubieda had fixed as well

**Review by ubieda — DISMISSED** [review]

**Review by teburd — DISMISSED** [review]

**Review by MaureenHelm — DISMISSED** [review]

**Review by jeppenodgaard — APPROVED** [review]

**Comment by ycsin** [comment]
Update the test with the new sensor shell output and now the CI pass

**Review by ubieda — APPROVED** [review]

**Review by jeppenodgaard — APPROVED** [review]

**Review by teburd — APPROVED** [review]

**Review by MaureenHelm — APPROVED** [review]

**Pull request merged**